great_expectations/self_check/util.py

```diff
@@ -99,6 +99,49 @@
     sqliteDialect = None
     SQLITE_TYPES = {}
 
+try:
+    import pybigquery.sqlalchemy_bigquery
+    import pybigquery.sqlalchemy_bigquery as BigQueryDialect
+
+    # Sometimes "pybigquery.sqlalchemy_bigquery" fails to self-register in certain environments, so we do it explicitly.
+    # (see https://stackoverflow.com/questions/53284762/nosuchmoduleerror-cant-load-plugin-sqlalchemy-dialectssnowflake)
+    sqlalchemy.dialects.registry.register(
+        "bigquery", "pybigquery.sqlalchemy_bigquery", "BigQueryDialect"
+    )
+    try:
+        getattr(pybigquery.sqlalchemy_bigquery, "INTEGER")
+        bigquery_types_tuple = {}
+        BIGQUERY_TYPES = {
+            "INTEGER": pybigquery.sqlalchemy_bigquery.INTEGER,
+            "NUMERIC": pybigquery.sqlalchemy_bigquery.NUMERIC,
+            "STRING": pybigquery.sqlalchemy_bigquery.STRING,
+            "BIGNUMERIC": pybigquery.sqlalchemy_bigquery.BIGNUMERIC,
+            "BYTES": pybigquery.sqlalchemy_bigquery.BYTES,
+            "BOOL": pybigquery.sqlalchemy_bigquery.BOOL,
+            "BOOLEAN": pybigquery.sqlalchemy_bigquery.BOOLEAN,
+            "TIMESTAMP": pybigquery.sqlalchemy_bigquery.TIMESTAMP,
+            "TIME": pybigquery.sqlalchemy_bigquery.TIME,
+            "FLOAT": pybigquery.sqlalchemy_bigquery.FLOAT,
+            "DATE": pybigquery.sqlalchemy_bigquery.DATE,
+            "DATETIME": pybigquery.sqlalchemy_bigquery.DATETIME,
+        }
+    except AttributeError:
+        # In older versions of the pybigquery driver, types were not exported, so we use a hack
+        logger.warning(
+            "Old pybigquery driver version detected. Consider upgrading to 0.4.14 or later."
+        )
+        from collections import namedtuple
+
+        BigQueryTypes = namedtuple(
+            "BigQueryTypes", sorted(pybigquery.sqlalchemy_bigquery._type_map)
+        )
+        bigquery_types_tuple = BigQueryTypes(**pybigquery.sqlalchemy_bigquery._type_map)
+except (ImportError, AttributeError):
+    bigquery_types_tuple = None
+    BigQueryDialect = None
+    pybigquery = None
+
```

Comment:
I don't have the requisite knowledge to fully grasp this distinction between old/new pybigquery drivers but I am always wary of `try/except` blocks nested within each other. 

Happy to accept this if you deem it appropriate but I did want to highlight it.

Reply:
Yea, this bugs me too. I'm going to block off some time this afternoon to see if I can attempt to detangle this... will report back

Reply:
If John signed off on this, it is alright by me. I'm always wary of `try/except` but it is more of an accepted pattern in Python. If this still bugs you, maybe put a ticket in the backlog? Definitely not necessary though.